src/plugins/topic_viewer/TopicViewer_TEST.cc

```diff
@@ -0,0 +1,193 @@
+
+    std::this_thread::sleep_for(std::chrono::milliseconds(300));
+
+    pub.Publish(msg);
+    pubInt.Publish(msgInt);
+
+    // ========== Load the Plugin ============
+    common::Console::SetVerbosity(4);
+
+    Application app(g_argc, g_argv);
+    app.AddPluginPath(std::string(PROJECT_BINARY_PATH) + "/lib");
+
+    // Load plugin
+    const char *pluginStr =
+      "<plugin filename=\"TopicViewer\">"
+        "<ignition-gui>"
+          "<title>Topic Viewer</title>"
+        "</ignition-gui>"
+      "</plugin>";
+
+    tinyxml2::XMLDocument pluginDoc;
+    EXPECT_EQ(tinyxml2::XML_SUCCESS, pluginDoc.Parse(pluginStr));
+
+    EXPECT_TRUE(app.LoadPlugin("TopicViewer",
+        pluginDoc.FirstChildElement("plugin")));
+
+    // Get main window
+    auto win = app.findChild<MainWindow *>();
+    EXPECT_NE(nullptr, win);
+
+    // Get plugin
+    auto plugins = win->findChildren<plugins::TopicViewer *>();
+    ASSERT_EQ(plugins.size(), 1);
+
+    auto plugin = plugins[0];
+    ASSERT_NE(nullptr, plugin);
+
+    // ============= Model =====================
+    auto model = plugin->Model();
+    ASSERT_NE(model, nullptr);
+
+    auto root = model->invisibleRootItem();
+    ASSERT_NE(model, nullptr);
+
+    ASSERT_EQ(root->hasChildren(), true);
+
+    bool foundCollision = false;
+    bool foundInt = false;
+
+    EXPECT_GE(root->rowCount(), 2);
+
+    // check plotable items
+    for (int i = 0; i < root->rowCount(); ++i)
+    {
+        auto child = root->child(i);
+
+        if (child->data(NAME_ROLE) == "/collision_topic")
+        {
+            foundCollision = true;
+
+            EXPECT_EQ(child->data(TYPE_ROLE), "ignition.msgs.Collision");
+            EXPECT_EQ(child->rowCount(), 8);
+
+            auto pose = child->child(5);
+            auto position = pose->child(3);
+            auto x = position->child(1);
+
+            EXPECT_EQ(x->data(NAME_ROLE), "x");
+            EXPECT_EQ(x->data(TYPE_ROLE), "double");
+            EXPECT_EQ(x->data(PATH_ROLE), "pose-position-x");
+            EXPECT_EQ(x->data(TOPIC_ROLE), "/collision_topic");
+            EXPECT_TRUE(x->data(PLOT_ROLE).toBool());
+        }
+        else if (child->data(NAME_ROLE) == "/int_topic")
+        {
+            foundInt = true;
+
+            EXPECT_EQ(child->data(TYPE_ROLE), "ignition.msgs.Int32");
+            EXPECT_EQ(child->rowCount(), 2);
+
+            auto data = child->child(1);
+
+            EXPECT_EQ(data->data(NAME_ROLE), "data");
+            EXPECT_EQ(data->data(TYPE_ROLE), "int32");
+            EXPECT_EQ(data->data(PATH_ROLE), "data");
+            EXPECT_EQ(data->data(TOPIC_ROLE), "/int_topic");
+            EXPECT_TRUE(data->data(PLOT_ROLE).toBool());
+        }
+        else
+        {
+            EXPECT_TRUE(false);
+        }
+    }
+
+    EXPECT_TRUE(foundCollision);
+    EXPECT_TRUE(foundInt);
+
+    // =========== Dynamic Adding / Removing =============
+
+    // Add
+    auto pubEcho = node.Advertise<msgs::StringMsg> ("/echo_topic");
```

Comment:
```suggestion
    auto pubEcho = node.Advertise<msgs::Collision> ("/echo_topic");
```

This is currently failing to publish because of the type mismatch:

```
Node::Publisher::Publish() Type mismatch.
        * Type advertised: ignition.msgs.StringMsg
        * Type published: ignition.msgs.Collision
```